netdata/netdata
# Add the ability to perform a data query using an offline node id

##### Summary
When a data query for an node id is received from the cloud, if the node is not found in memory (linked to a host structure in memory), the query fails. 
With this PR a temporary host structure is allocated so that the query can be performed. This will  allow a data query to be done using any of the following

1. /host/hostname/api/v1/data
1. /host/machine_guid/api/v1/data
1. /host/node_id/api/v1/data

##### Test Plan
- Setup parent - child and claim parent to the cloud
- Start parent and parent and visit the cloud dashboard
- Notice queries in the parent `access.log` 
- Stop the parent and the child
- Start only the parent and return to the dashboard
  - Notice 404 errors in the parents `access.log` (failed queries attempting to access the child node)
- Redo tests after applying the PR
  


##### Additional Information


<details> <summary>For users: How does this change affect me?</summary>
  
</details>
